reference/7.0/Microsoft.PowerShell.Core/About/about_Calculated_Properties.md

```diff
@@ -0,0 +1,523 @@
+---
+Locale: en-US
+ms.date: 08/07/2020
+online version: https://docs.microsoft.com/powershell/module/microsoft.powershell.core/about/about_calculated_properties?view=powershell-7&WT.mc_id=ps-gethelp
+schema: 2.0.0
+title: about_Calculated_Properties
+---
+# About calculated properties
+
+## Short Description
+
+Calculated properties allow you to dynamically add properties and formatting to
+objects in the pipeline.
+
+## Long Description
+
+There are several cmdlets that operate on properties of objects. PowerShell
+provides a method to create calculated properties. Calculated properties
+dynamically create new properties that are added to the output object and can
+provide custom formatting for an existing property.
+
+Each cmdlet that supports calculated properties has a specific set of
+attributes that describe the property. The attributes define the name of the
+property, an expression to calculate the value, and optional formatting
+information. A calculated property is defined using a
+[hashtable](about_hash_tables.md) containing these attributes as key-value
+pairs.
```

Comment:
I think approaching this subject from the angle of the properties on the output objects themselves confuses the material. You can see that in this first paragraph where we're forced to overload "properties" to mean both the parameter and the property on the output object.

My feeling is that it would be better to write the article *about the common parameters* that generate calculated properties, rather than trying to write directly about the properties themselves.

Otherwise:

> There are several cmdlets that operate on properties of objects.

This phrase doesn't tell me anything about what you want me to know, and is too general for me to really understand what you're getting at.

> PowerShell provides a method to create calculated properties.

This doesn't relate strongly to the previous statement, refers to "a method" without qualifying it in any way, and uses "calculated properties" as the object of the sentence without introducing or defining the concept.

> Calculated properties dynamically create new properties that are added to the output object and can provide custom formatting for an existing property.

The word properties is used twice here to mean different things, we depend on the also unintroduced concept of the "output object" and within the same sentence launch into the subject of formatting.

> Each cmdlet that supports calculated properties has a specific set of
attributes that describe the property.

We're now back to cmdlets, from objects and output objects, but refer to "the property" without the definiteness the article implies. Plus we've introduced attributes.

And so on.

Instead, addressing the subject from the entry point of the parameter:

> A number of PowerShell cmdlets that transform, aggregate or process input objects into output objects provide parameters that allow the addition of new properties to those output objects.
> These property-adding parameters can be used to generate new *calculated properties* on output objects based on the values of input objects.
> In particular, a property-adding parameter takes a hashtable value which can be used to set the name, calculating scriptblock expression and formatting information for a property it generates on an output object.

Reply:
@rjmholt I reworked the intro. Let me know if that reads better.